graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/functions/json/JsonUtils.java

```diff
@@ -61,12 +63,41 @@ public static JsonNode extractJson(
         ObjectNode resultRoot = mapper.createObjectNode();
         for (Map.Entry<String, Object> mapEntry : json.entrySet()) {
             for (Entry entry : parseValue(mapEntry.getKey(), mapEntry.getValue(), mapper, extractFlags)) {
-                resultRoot.put(entry.key(), entry.value().toString());
+                if (stringify) {
+                    resultRoot.put(entry.key(), entry.value().toString());
+                } else {
+                    putNodeWithType(resultRoot, entry.key(), entry.value());
```

Comment:
```suggestion
                    resultRoot.putPOJO(entry.key(), entry.value());
```
does the same, no?

Reply:
Thx - should have looked a bit harder for this.